### PR TITLE
Feature/add contracts to profile

### PR DESCRIPTION
**Pull request by mcromie92**
Some files have overlapped from my other branch, please merge previous branch first

### CODE REVIEW

File: web-app/src/pages/Profile/container.js [hunk 1]
```diff
@@ -56,7 +60,20 @@ const ProfileContainer = Wrapped =>
             });
           });
         });
-      }
+      } 
+      if (
+        this.props.userDetails.employeeId !== null &&
+        this.state.contracts !== null
+      )
+        {
```
Comment (ghost) [inline web-app/src/pages/Profile/container.js]: Tabbing seems to be a bit off.

File: web-app/src/pages/Profile/container.js [hunk 2]
```diff
@@ -24,12 +26,14 @@ const ProfileContainer = Wrapped =>
...
         daysPending: null,
         selectedHoliday: {},
+        contracts: [],
       };
     }
 
     componentDidMount() {
       // Required in case the user navigates away from the page, then back.
       this.setState({ holidays: null });
+      this.setState( { contracts: [] });
```
Comment (ghost) [inline web-app/src/pages/Profile/container.js]: I'd just make sure the spacing etc is the same as the others.

File: web-app/src/pages/Profile/container.js (hunk 1 above)
Comment (ghost) [reply web-app/src/pages/Profile/container.js]: So what happens if the conditions fails, it just doesn't run `getContractsByEmployeeId` is that expected behaviour?

**Review by ghost — CHANGES_REQUESTED** [review]
Just a few things, tabbing needs just to be looked at closer.

File: web-app/src/pages/Profile/container.js [hunk 3]
```diff
@@ -56,7 +60,20 @@ const ProfileContainer = Wrapped =>
...
           });
         });
-      }
+      } 
+      if (
+        this.props.userDetails.employeeId !== null &&
+        this.state.contracts !== null
+      )
+        {   
+        getContractsByEmployeeId(this.props.userDetails.employeeId).then(response => {
```
Comment (William-Unosquare) [inline web-app/src/pages/Profile/container.js]: We already chatted about this, but I'm just making sure this isn't merged in yet by accident. Basically, we can't update state in `ComponentDidUpdate` or you'll be stuck in a loop.

**Review by William-Unosquare — CHANGES_REQUESTED** [review]

**Comment by William-Unosquare** [comment]
As for tabbing, if you're using VSCode, get the extension [prettier](https://prettier.io/). It'll keep your code style in check automatically every time you save the file. It's a great time saver.

You'll also need to set these preferences up (File -> Preferences -> Settings)
```
{
  "editor.tabSize": 2,
  "prettier.trailingComma": "all",
  "prettier.singleQuote": true,
  "prettier.eslintIntegration": true,
}
```

**Comment by mcromie92** [comment]
@girwin575 fixed the tabbing and spacing, installed prettier now too :)
@William-Unosquare I have updated the function so it is not called inside the componentDidUpdate to remove the loop

**Review by William-Unosquare — APPROVED** [review]
Looks great, approved!

**Pull request merged**